Feature/리포트 API 인터페이스

koosco opened the pull request:
## Related Issue 🔗
- closes #60 

## Summary 📝
* 리포트 api 인터페이스
* local 설정 파일 추가 -> application-local.yaml
  * (일단 dev와 동일하게 설정해두었고 추후 논의를 통해 dev가 postgresql을 바라보도록 수정하면 될 것 같습니다)
  * 로컬 환경에서 swagger 테스트를 위해 추가
* AuthController local login api 추가
  * 실제 처리는 SecurityFilter에서 처리
  * swagger에 노출시키기 위해 controller에 추가

## Changes ✨
<!-- 구체적인 코드/설정/구조 변경 사항을 나열해주세요 -->

## Why we need 💡
<!-- 이 변경이 필요한 이유와 배경을 설명해주세요 -->

## Test 🧪
```
{
    "status": 200,
    "message": "요청 데이터가 성공적으로 조회되었습니다.",
    "data": {
        "updatedAt": "2025-09-23T16:39:52.059834",
        "poo": {
            "score": 85.5,
            "summary": {
                "image": "https://kr.object.ncloudstorage.com/depromeet-dev-static-resources/very_bad_colon.png",
                "backgroundColors": [
                    "#A4141E",
                    "#FF535F"
                ],
                "caption": "화가 잔뜩 난 대장",
                "message": "전문가 상담이 필요해요"
            },
            "items": [
                {
                    "occurredAt": "2025-09-23T16:39:52.070009",
                    "message": "전문가의 상담이 필요해요. 복통이\n매우 심했다면 단순한 식사 문제를\n넘어서 장염이나 자극적인 음식 섭취",
                    "color": "DARK_BROWN",
                    "shape": "PORRIDGE",
                    "duration": 10,
                    "pain": 78.0,
                    "note": "배가 너무 아팠어요"
                }
            ]
        },
        "food": {
            "message": "맵고 자극적인 음식이 장을 자극했을 수 있어요",
            "items": [
                {
                    "occurredAt": "2025-09-22T16:39:52.070089",
                    "meals": [
                        {
                            "mealTime": "BREAKFAST",
                            "dangerous": false,
                            "foods": [
                                "토스트",
                                "커피",
                                "계란"
                            ]
                        },
                        {
                            "mealTime": "LUNCH",
                            "dangerous": true,
                            "foods": [
                                "떡볶이",
                                "순대",
                                "튀김"
                            ]
                        },
                        {
                            "mealTime": "DINNER",
                            "dangerous": false,
                            "foods": [
                                "샐러드",
                                "닭가슴살",
                                "고구마"
                            ]
                        },
                        {
                            "mealTime": "SNACK",
                            "dangerous": false,
                            "foods": [
                                "아이스크림",
                                "과자"
                            ]
                        }
                    ]
                },
                {
                    "occurredAt": "2025-09-23T16:39:52.070163",
                    "meals": [
                        {
                            "mealTime": "BREAKFAST",
                            "dangerous": false,
                            "foods": [
                                "우유",
                                "바나나",
                                "시리얼"
                            ]
                        },
                        {
                            "mealTime": "LUNCH",
                            "dangerous": false,
                            "foods": [
                                "된장찌개",
                                "고등어구이",
                                "김치"
                            ]
                        },
                        {
                            "mealTime": "DINNER",
                            "dangerous": false,
                            "foods": [
                                "닭가슴살",
                                "샐러드",
                                "고구마"
                            ]
                        },
                        {
                            "mealTime": "SNACK",
                            "dangerous": false,
                            "foods": [
                                "견과류"
                            ]
                        }
                    ]
                }
            ]
        },
        "water": {
            "message": "장이 말라가고 있어요! 물 섭취량을 늘려야 해요",
            "items": [
                {
                    "name": "STANDARD",
                    "value": 2000.0,
                    "color": "#4E5560",
                    "level": "STANDARD"
                },
                {
                    "name": "YESTERDAY",
                    "value": 500.0,
                    "color": "#F4B005",
                    "level": "MEDIUM"
                },
                {
                    "name": "TODAY",
                    "value": 300.0,
                    "color": "#F13A49",
                    "level": "LOW"
                }
            ]
        },
        "stress": {
            "message": "스트레스 관리가 필요해요.\n가벼운 산책이나 명상은 어때요?",
            "image": "http://dummy_stress_image.png"
        },
        "suggestion": {
            "message": "장 상태를 개선하려면 이런 습관을 추천해요",
            "items": [
                {
                    "image": "https://kr.object.ncloudstorage.com/depromeet-dev-static-resources/suggestion_water.png",
                    "title": "물 섭취량을 더 늘려보세요",
                    "content": "하루 권장 물 섭취량은 성인 기준 2L 예요"
                },
                {
                    "image": "https://kr.object.ncloudstorage.com/depromeet-dev-static-resources/suggestion_food.png",
                    "title": "충분한 식이섬유가 중요해요",
                    "content": "과일과 채소를 섭취하면 좋은 흐름이 유지돼요"
                },
                {
                    "image": "https://kr.object.ncloudstorage.com/depromeet-dev-static-resources/suggestion_note.png",
                    "title": "지속적으로 배변을 기록해요",
                    "content": "배변이 잘 되는 나만의 루틴을 만들 수 있어요"
                }
            ]
        }
    }
}
```

## Trouble Shooting 🐛
<!-- 어떤 위험이나 장애를 발견했는지 적어주세요 -->

## ScreenShot 📷
<!-- 스크린샷 첨부 -->

## To Reviewers 🙏
<!-- 리뷰어들에게 물어볼 점, 할 말 등을 적어주세요 -->

## CC 👥
<!-- 함께 알림을 받아야 하는 사람을 멘션해주세요 (@username) -->

## Anything else? 💬
<!-- 추가로 공유하고 싶은 내용, 참고 자료 링크 등을 작성해주세요 -->


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- 신규 기능
  - 로컬 로그인 API 추가: POST /api/v1/auth/login (JSON 요청/응답)
  - 일일 리포트 조회 API 추가: GET /api/v1/reports/daily (인증 필요). 배변·식단·수분·스트레스·추천 섹션으로 구성된 요약 리포트 제공. 날짜/시간 파라미터 미지정 시 현재 시각 사용
- 작업
  - 실행 프로필 정리: 기본 프로필을 local로 변경, dev 설정 분리
  - 데이터베이스 및 마이그레이션 설정 개선(환경변수 기반), CORS/Swagger 환경 변수화
  - Makefile 업데이트 및 Postgres 전용 설정 파일 정리
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Commit:
[#60] feat : daily report dummy api 추가

Commit:
[#60] config(application-local.yaml) : local 설정 파일 추가

Commit:
[#60] fix(AuthController) : swagger에 로컬 로그인이 렌더링되도록 더미 api 추가

Commit:
[#60] fix : test용 설정값 추가

ihnokim commented on src/main/java/depromeet/lessonfour/server/report/app/dto/response/GetDailyReportResponseDto.java:
```diff
@@ -0,0 +1,61 @@
+  public record PooDailyReport(double score, PooSummary summary, List<PooReportItem> items) {}
+
+  public record PooSummary(
+      String image, String firstGradient, String secondGradient, String caption, String message) {}
```
색상은 gradient이므로 기존처럼 List[string] 형식으로 반환하는 것이 좋을 것 같습니다! (2개 이상의 색상을 사용할 수 있음)
field 명은 colors 또는 backgroundColors로 기존처럼 사용하는 것이 어떨까요?

koosco replied:
gradient 색상 순서가 중요할 것 같은데 리스트로 내려줘도 상관없을까요?

ihnokim commented on src/main/java/depromeet/lessonfour/server/report/app/mapper/DailyReportMapper.java:
```diff
@@ -0,0 +1,109 @@
+                LocalDateTime.now().minusDays(1),
+                List.of(
+                    new FoodReportMeal(MealTime.BREAKFAST, false, List.of("토스트", "커피", "계란")),
+                    new FoodReportMeal(MealTime.LUNCH, true, List.of("매운 떡볶이", "순대", "튀김")),
```
Minor하긴 한데, food 이름을 공백없이 제공할 예정이라, "매운 떡볶이" -> "매운떡볶이" or "떡볶이"로 바꿔주실 수 있나요? :)

Commit:
[#60] fix : local flyway 활성화

Commit:
[#60] fix : dev 환경 ddl-auto validate

Commit:
[#60] fix : Refactor date handling and adjust PooSummary structure

Update `dateTime` parameter usage in `GetDailyReportUseCase` to improve date precision and refactor `PooSummary` to use a list for background colors. Minor text and structure adjustments are included for improved clarity and consistency in report generation.

ihnokim commented on src/main/resources/application-dev.yaml:
```diff
@@ -9,22 +12,29 @@ spring:
   jpa:
     hibernate:
-      ddl-auto: update
+      ddl-auto: validate
```
이렇게 설정해두면 h2로 db 연결돼서 flyway migration fail나지 않나요!?

ihnokim replied:
이전에는 UUID 때문에 오류나는 걸로 기억하는데, UUID가 빠지면서 정상화되어서 바꾼 것인지 궁금합니다 ㅎㅎ

koosco replied:
이 부분은 제가 테스트가 부족했던 것 같습니다! 다시 확인해보니 h2에서 uuid 함수가 달라 fail이 발생하네요..
dev profile에서 local postgresql을 연결하도록 수정해두었습니다.
local profile도 flyway를 비활성화하고 ddl-auto: update로 수정하였습니다

ihnokim commented:
고생 많으셨습니다! comment 확인해주시고, test 부분에 최종 json 예시 올려주시면 감사드리겠습니다 🙏

Commit:
[#60] config : dev, local db connection 정보 수정

Commit:
[#60] fix : dummy data 수정

Commit:
[#60] fix : datetime default 값 추가

ihnokim commented on src/main/resources/application.yaml:
```diff
@@ -6,10 +6,6 @@ spring:
       pg:
```
profile이름들을 변경했으니 이 group 부분도 수정해야 할 듯 합니다!

Commit:
Merge branch 'develop' into feat/#60

Commit:
Merge branch 'develop' into feat/#60

Commit:
[#60] config : pg profile 삭제
